src/plugins/explore/public/application/utils/state_management/actions/query_editor/run_query/run_query.ts

```diff
@@ -13,17 +13,22 @@ import { detectAndSetOptimalTab } from '../../detect_optimal_tab';
 /**
  * This is called when you want to run the query
  */
-export const runQueryActionCreator = (services: ExploreServices, query?: string) => async (
-  dispatch: AppDispatch
-) => {
+export const runQueryActionCreator = (
+  services: ExploreServices,
+  query?: string,
```

Comment:
can we change this to an object prop instead of 3 arguments?

- its hard to read (what is false in `runQueryActionCreator(services, query, false)`?)
- what if query is undefined but `isUpdate` is true? Then you need to do `runQueryActionCreator(services, undefined, false)`